### PR TITLE
feat(#470): add support for dataframe with multiple values

### LINKED ISSUE #470: feat: add support for dataframe with multiple values
We want to support something similar to what was previously offered by   `df_with_multiple_values_to_named_dataframes` followed by `PredictorGroupSpec(...).make_combinations()` if users want to input a dataframe with multiple columns with the same timestamp (e.g. embedded text). 

**Proposed solution**
Infer the `value_col_name` in `ValueFrame` as being a `Sequence[str]`containing all columns that are not `entity_id` or `timestamp`. Rewrite masking, aggregation, and col checkers to handle `value_col_names` now being a Sequence

### PR DESCRIPTION
**Pull request by HLasse**
Infer value columns and process at once

feat(#470): add support for dataframe with multiple values

Fixes #470

### CODE REVIEW

**Comment by HLasse** [comment]
Current dependencies on/for this PR:
* **#481** <a href="https://app.graphite.dev/github/pr/Aarhus-Psychiatry-Research/timeseriesflattener/481?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/Aarhus-Psychiatry-Research/timeseriesflattener/481?utm_source=stack-comment).

File: .vscode/settings.json [hunk 1]
```diff
@@ -5,5 +5,6 @@
     "python.testing.unittestEnabled": false,
     "python.testing.pytestEnabled": true,
     "python.analysis.typeCheckingMode": "standard",
-    "python.analysis.diagnosticMode": "workspace"
+    "python.analysis.diagnosticMode": "workspace",
+    "CodeGPT.apiKey": "OpenAI"
```
Comment (MartinBernstorff) [inline .vscode/settings.json:9]: ... 😉

File: src/timeseriesflattenerv2/feature_specs/timedelta.py [hunk 2]
```diff
@@ -36,10 +36,10 @@ def __post_init__(self):
             init_df=self.init_frame.df.rename(
                 {self.init_frame.value_timestamp_col_name: self.output_name}
             ),
-            value_col_name=self.output_name,
             entity_id_col_name=self.init_frame.entity_id_col_name,
             value_timestamp_col_name=self.output_name,
         )
+        self.value_frame.value_col_names = [self.output_name]
```
Comment (MartinBernstorff) [inline src/timeseriesflattenerv2/feature_specs/timedelta.py:42]: Ensure this is checked somewhere.

File: src/timeseriesflattenerv2/spec_processors/static.py [hunk 3]
```diff
@@ -12,15 +12,16 @@
 def process_static_spec(
     spec: StaticSpec, predictiontime_frame: PredictionTimeFrame
 ) -> ProcessedFrame:
-    new_col_name = (
-        f"{spec.column_prefix}_{spec.value_frame.value_col_name}_fallback_{spec.fallback}"
-    )
+    new_col_names = [
```
Comment (MartinBernstorff) [inline src/timeseriesflattenerv2/spec_processors/static.py:15]: Turn this into `old2new_colname`? Should make the rename and select statements much easier to read.

File: src/timeseriesflattenerv2/spec_processors/temporal.py [hunk 4]
```diff
@@ -53,34 +80,46 @@ def _mask_outside_lookperiod(
...
+                value_col_name in col for value_col_name in masked_frame.value_col_names
+            )
+        )
         .map(lambda old_name: (old_name, f"{old_name}_fallback_{fallback}"))
     )
     rename_mapping = dict(value_columns)
 
     with_fallback = grouped_frame.with_columns(
-        cs.contains(masked_frame.value_col_name).fill_null(fallback)
+        cs.contains(masked_frame.value_col_names).fill_null(fallback)
```
Comment (MartinBernstorff) [inline src/timeseriesflattenerv2/spec_processors/temporal.py:122]: I have a hard time reasoning well about this code, and AFAICT, it's untested. We need a test with multiple input columns and multiple aggregators 😊

**Review by MartinBernstorff — COMMENTED** [review]
Looks great Lasse! A few comments 👍

File: src/timeseriesflattenerv2/feature_specs/timedelta.py (hunk 2 above)
Comment (HLasse) [reply src/timeseriesflattenerv2/feature_specs/timedelta.py:42]: The `value_timestamp_col_name` is renamed to this above (that value was checked), so it exists

File: src/timeseriesflattenerv2/spec_processors/temporal.py (hunk 4 above)
Comment (HLasse) [reply src/timeseriesflattenerv2/spec_processors/temporal.py:122]: Added, it works 🙌

**Pull request merged**